controllers/storageclaim_controller.go

```diff
@@ -599,27 +612,36 @@ func (r *StorageClaimReconciler) createOrReplaceVolumeSnapshotClass(volumeSnapsh
 		return fmt.Errorf("failed to get VolumeSnapshotClass: %v", err)
 	}
 
-	// If present then compare the existing VolumeSnapshotClass parameters with
-	// the received VolumeSnapshotClass parameters, and only proceed if they differ.
-	if reflect.DeepEqual(existing.Parameters, volumeSnapshotClass.Parameters) {
+	// If present then compare the existing VolumeSnapshotClass parameters and labels with
+	// the received VolumeSnapshotClass parameters and labels, and only proceed if they differ.
+	labelsEqual := reflect.DeepEqual(existing.Labels, volumeSnapshotClass.Labels)
+	if reflect.DeepEqual(existing.Parameters, volumeSnapshotClass.Parameters) && labelsEqual {
```

Comment:
Let's make the code more readable by aligning both the params and label equality checks
Also, we are setting annotations so we should also consider checking equality there as well

``suggestion
	areLabelsEqual := eflect.DeepEqual(existing.Labels, volumeSnapshotClass.Labels)
	areAnnotationsEqual := eflect.DeepEqual(existing.Annotations, volumeSnapshotClass.Annotations)
	areParamsEqual := reflect.DeepEqual(existing.Parameters, volumeSnapshotClass.Parameters)
	if areLabelsEqual && areAnnotationsEqual && areParamsEqual {